### ISSUE TITLE
Some question about the loss function.

### ISSUE THREAD

**Issue by Luchixiang** [locuslab/ect#9]
Hi. I noticed a discrepancy between the code with the paper about the ect loss function. 
The first one is adaptive weight, in the paper, it says the weight is 1 / sqrt((f(x_t) - f(x_r)) ^2) + c, but in the code the weight is 1/ (t-r).  Do these implementations perform similarly?

Another question is about the r/t. In the paper, it says r/t = 1 - 1/ q^a*n(t) while in the code r/t = 1 - 1/ q^(a+1) * n(t). Seems like in the code, it ignores the same objective as the diffusion model (when a = 0).  Am I misunderstanding something or it is okay to ignore it since we resume from a pretrained diffusion model?

**Comment by Gsunshine** [comment]
Hi @Luchixiang ,

Apologies for the delayed response. I've been occupied with the NeurIPS rebuttal.

1. We use both adaptive weighting (which can be based on metrics or combined with other metrics) and timestep weighting (which we refer to as regular weighting). Having the efficiency of ECT, you can choose the one that best fits your requirements through quick experiments.

2. Regarding the r/t ratio, you can skip this stage if you're starting from a pretrained model.

Best of luck with your work!

Zhengyang
